### PR TITLE
Do not use multiple unnamed structs inside of unions

### PR DESCRIPTION
**Pull request by ret2libc**
Related to #604 .

I could not completely avoid naming some of the structures, as it seems GCC 4 (at least as provided on CentOS 6) has problems with multiple unnamed structs inside a union.

What do you think about this @maxbrunsfeld  ?

### CODE REVIEW

File: lib/include/tree_sitter/parser.h [hunk 1]
```diff
@@ -62,13 +62,13 @@ typedef struct {
       TSStateId state;
       bool extra : 1;
       bool repetition : 1;
-    };
+    } status;
```
Comment (maxbrunsfeld) [inline lib/include/tree_sitter/parser.h:65]: I think this field should be called `shift`. "Shift" and "reduce" are the two main types of actions.

File: lib/include/tree_sitter/parser.h [hunk 2]
```diff
@@ -167,22 +167,22 @@ struct TSLanguage {
...
 
-#define SHIFT(state_value)              \
-  {                                     \
-    {                                   \
-      .type = TSParseActionTypeShift,   \
-      .params = {.state = state_value}, \
-    }                                   \
+#define SHIFT(state_value)                \
+  {                                       \
+    {                                     \
```
Comment (maxbrunsfeld) [inline lib/include/tree_sitter/parser.h:172]: I think we should use the field name here, for consistency with the `REDUCE` macro.

File: lib/include/tree_sitter/parser.h (hunk 1 above)
Comment (ret2libc) [reply lib/include/tree_sitter/parser.h:65]: yeah sure. I didn't know how to name it as I'm not that familiar with tree-sitter internals

**Comment by maxbrunsfeld** [comment]
Thanks!

**Pull request merged**